### ISSUE TITLE
Agent creation fails for Linux machines

### ISSUE THREAD

**Issue by joestr** [gocd/azure-elastic-agent-plugin#23]
The preparation for Linux VM is flawed.

While in the prepare script JDK 13 gets downloaded and installed in the start script references a JDK 12 installation.

__Prepare script__

https://github.com/gocd/azure-elastic-agent-plugin/blob/73374433c593102f52eb9cf4dc36bbb0f4a8cb14/src/main/resources/post_provision_script.template.ftlh#L5
https://github.com/gocd/azure-elastic-agent-plugin/blob/73374433c593102f52eb9cf4dc36bbb0f4a8cb14/src/main/resources/post_provision_script.template.ftlh#L15

__Start script__

https://github.com/gocd/azure-elastic-agent-plugin/blob/73374433c593102f52eb9cf4dc36bbb0f4a8cb14/src/main/resources/linux_start_go_agent.template.flth#L4

**Comment by chadlwilson** [comment]
Thanks, feel free to submit a PR. Seems a trivial mistake able to be worked around by setting JAVA_HOME?

I think it's fair to say that this plugin is basically abandoned at the moment. It has never been updated to the latest elastic agent plugin API so I would be a little surprised if it works on current GoCD versions.

Personally i don't have an Azure account or Azure expertise so if you have a need for this or are using this currently and would like to whip it into shape, help would be welcome.

**Comment by joestr** [comment]
Hi,

Thanks for the response!

I mainly use Azure because they are rather cheap and can spin up Windows VMs comparatively fast.

The latest version of this plugin uses a [hacky solution](https://github.com/gocd/azure-elastic-agent-plugin/blob/master/docs/ELASTIC_PROFILE_CONFIGURATION.md) to function properly.

I created [a fork](https://github.com/joestr/azure-elastic-agent-plugin) where I fixed the scripts. Tackling the API change would take some time since I have to get familiar with the API.
